backend/app/api/v1/analytics.py

```
@@ -307,6 +307,52 @@ async def get_agent_performance_metrics(
 
 # Dashboard Endpoints
 
+@router.get("/dashboard/overview", response_model=Dict[str, Any])
+async def get_dashboard_overview(
+    period_hours: int = Query(24, ge=1, le=8760, description="Time period in hours"),
+    current_user: User = Depends(get_current_user)
```

Ruff (0.12.2)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)